handel/tests/mod.rs

```diff
@@ -300,7 +296,7 @@ async fn it_can_aggregate() {
     }
 
     loop {
-        match tokio::time::timeout_at(deadline, aggregation.next()).await {
+        match nimiq_time::timeout(timeout, aggregation.next()).await {
```

Comment:
This changes the timeout logic from 300 ms \* contributors / 7 for the entire thing to 300 ms \* contributors / 7 per loop.

Reply:
Do you mean the variable was copied wrong (`deadline` => `timeout`), or the new `timeout` method behaves differently than `timeout_at` did?

Reply:
I think I mean neither.

Previously, the `timeout_at` function made sure that the entire loop did not take longer than `timeout_ms`. Now, the usage of the `timeout` function only makes sure that each loop *iteration* takes no longer than `timeout`.

I.e. previously the whole loop was limited in time, now each *iteration* can take as long as the whole loop previously.

Not sure if that made it clearer…

Reply:
Yes, I understand now.

OK, so since we don't have a `timeout_at` anymore, we need to make the `timeout` duration dynamic. Calculate it from the difference of `deadline - <now>`. Would that be right?

Reply:
Yes, that would work.

Reply:
Done, check out https://github.com/nimiq/core-rs-albatross/pull/2318/commits/a3fb6060120bcb67e01b4e5943f3b471be1462cb